USAGE.md

```diff
@@ -167,6 +167,66 @@ provider = function(_, _, opts)
 end
 ```
 
+#### Component short provider
+
+Feline has an automatic smart truncation system where components can be automatically truncated if the statusline doesn't fit within the window. In order to make use of this truncation system, you have to define the `short_provider` component value.
+
+`short_provider` works just like the `provider` value, but is activated only when the component is being truncated due to the statusline not fitting within the window. `short_provider` is independent from the `provider` value so it can be a different provider altogether, or it can be a shortened version of the same provider or the same provider but with a different `opts` value. For example:
+
+```lua
+-- In this component, short provider uses same provider but with different opts
+local file_info_component = {
+    provider = {
+        name = 'file_info',
+        opts = {
+            type = 'full-path'
+        }
+    },
+    short_provider = {
+        name = 'file_info',
+        opts = {
+            type = 'short-path'
+        }
+    }
+}
+
+-- Short provider can also be an independent value / function
+local my_component = {
+    provider = 'loooooooooooooooong',
+    short_provider = 'short'
+}
+```
+
+Feline doesn't set `short_provider` to any component by default, so it must be provided manually.
+
+#### Hiding components during truncation
```

Comment:
This will get generated as `feline-Hiding-components-during-truncation`, so maybe simplify the heading to `Hide truncated component`